Modules/signalmodule.c

```diff
@@ -104,37 +104,58 @@ class sigset_t_converter(CConverter):
    may not be the thread that received the signal.
 */
 
-static volatile struct {
+struct _signal_handler_state {
     _Py_atomic_int tripped;
     PyObject *func;
-} Handlers[NSIG];
+};
 
 #ifdef MS_WINDOWS
-#define INVALID_FD ((SOCKET_T)-1)
+#  define INVALID_FD ((SOCKET_T)-1)
+#else
+#  define INVALID_FD (-1)
+#endif
 
-static volatile struct {
-    SOCKET_T fd;
-    int warn_on_full_buffer;
-    int use_send;
-} wakeup = {.fd = INVALID_FD, .warn_on_full_buffer = 1, .use_send = 0};
+struct _signal_wakeup_state {
+#ifdef MS_WINDOWS
+    volatile SOCKET_T fd;
 #else
-#define INVALID_FD (-1)
-static volatile struct {
     sig_atomic_t fd;
-    int warn_on_full_buffer;
-} wakeup = {.fd = INVALID_FD, .warn_on_full_buffer = 1};
 #endif
+    volatile int warn_on_full_buffer;
+#ifdef MS_WINDOWS
+    volatile int use_send;
+#endif
+};
 
-/* Speed up sigcheck() when none tripped */
-static _Py_atomic_int is_tripped;
-
-static PyObject *DefaultHandler;
-static PyObject *IgnoreHandler;
-static PyObject *IntHandler;
+struct _signal_state {
+    struct _signal_handler_state handlers[NSIG];
+    struct _signal_wakeup_state wakeup;
+    // Speed up sigcheck() when none tripped
+    _Py_atomic_int is_tripped;
+    // Integer singleton equal to SIG_DFL
+    PyObject *default_handler;
+    // Integer singleton equal to SIG_IGN
+    PyObject *ignore_handler;
+    // signal.default_int_handler method
+    PyObject *int_handler;
+#ifdef MS_WINDOWS
+    HANDLE sigint_event;
+#endif
+};
 
+static struct _signal_state global_signal_state = {
+    .wakeup = {
+        .fd = INVALID_FD,
+        .warn_on_full_buffer = 1,
+#ifdef MS_WINDOWS
+        .use_send = 0,
+#endif
+    },
 #ifdef MS_WINDOWS
-static HANDLE sigint_event = NULL;
+    .sigint_event = NULL,
 #endif
+};
+
 
 #ifdef HAVE_GETITIMER
 static PyObject *ItimerError;
```

Comment:
This variable is missed.

Reply:
Right, I noticed when I finished my PR :-) I added it.